src/command_executor.cc

```diff
@@ -17,19 +17,37 @@
 
 void CommandExecutor::OutputRedirect(
     const CommandParseResult &command_parse_result) {
+  int fd_out;
   if (command_parse_result.output_is_append) {
-    int fd_out = utils::SystemCallExitOnFailed(
+    fd_out = utils::SystemCallExitOnFailed(
         open(command_parse_result.output_redirect_file.c_str(),
              O_WRONLY | O_APPEND | O_CREAT, 0664));
-    utils::SystemCallExitOnFailed(dup2(fd_out, STDOUT_FILENO));
-    close(fd_out);
   } else {
-    int fd_out = utils::SystemCallExitOnFailed(
+    fd_out = utils::SystemCallExitOnFailed(
         open(command_parse_result.output_redirect_file.c_str(),
              O_WRONLY | O_TRUNC | O_CREAT, 0664));
-    utils::SystemCallExitOnFailed(dup2(fd_out, STDOUT_FILENO));
-    close(fd_out);
   }
+  utils::SystemCallExitOnFailed(dup2(fd_out, STDOUT_FILENO));
+  if (command_parse_result.is_error_redirect) {
```

Comment:
What does it mean for `is_error_redirect`, I can't get the meaning of it from the name. Do you mean redirect stderr to stdout?

Reply:
Looks strange, this method named OutputRedirect, and you redirect stderr.